ParabolInc/parabol
# chore: Add more Atlassian logging

Mainly associate the logs with the traces so it's possible to check which GraphQL request caused a certain debug output.
I needed to shuffle some stuff from `AtlassianManager` to `AtlassianServerManager` because `AtlassianManager` lives in `parabol-client` and we don't have the tracing stuff there. That's a cleaner solution anyways because we cannot call the API from the client because we wouldn't expose the secrets there.

## Demo

https://www.loom.com/share/0625c83f3d1845159d89db0253ba7da1?sid=fbbbd4ed-3719-470c-8fd5-f60ad2aeb9c9 🔒 

## Testing scenarios

- see demo
- check you have the correct datadog variables set (there is a [PR](https://github.com/ParabolInc/action-devops/pull/186/files#diff-8a86bc8dcf3eb7b34043c2b8509fec32fff3680a9c9067de3135e48216d974fe) 🔒) for these
- additionally add
  ```
  DD_LOGS_ENABLED='true'
  DD_LOGS_INJECTION='true'
  ```
- restart local datadog agent
  ```
  yarn db:stop ; yarn db:start
  ```
- run the dev build
  ```
  yarn dev
  ```
- check the agent status and the logs in DD
  ```
  docker exec -it docker-datadog-1 agent status
  ```
- add debug output to JiraServerManager::getAllProjects https://github.com/ParabolInc/parabol/blob/630d525434aa6863f92f508b4ec03d72655ea0da/packages/server/utils/AtlassianServerManager.ts#L402
  ```
  log('GEORG was here')
  ```
  make sure to only use `log` not `console.log`
- trigger the function call by integrating with atlassian, opening the task footer menu and searching for some non-existent project
- see the log message and trace connected in datadog

## Final checklist
